vhost/src/vhost_user/backend_req_handler.rs

```diff
@@ -563,6 +569,27 @@ impl<S: VhostUserBackendReqHandler> BackendReqHandler<S> {
                 let res = self.set_backend_req_fd(files);
                 self.send_ack_message(&hdr, res)?;
             }
+            Ok(FrontendReq::GET_SHARED_OBJECT) => {
+                self.check_proto_feature(VhostUserProtocolFeatures::SHARED_OBJECT)?;
+                self.check_request_size(&hdr, size, hdr.get_size() as usize)?;
+                let msg = self.extract_request_body::<VhostUserSharedMsg>(&hdr, size, &buf)?;
+                let res = self.backend.get_shared_object(msg);
+                let reply_hdr = self.new_reply_header::<VhostUserEmpty>(&hdr, 0)?;
+
+                match res {
```

Comment:
`res` is difficult to understand and also not needed, so I'd suggest to remove it and call `get_shared_object` here:

```diff
diff --git a/vhost/src/vhost_user/backend_req_handler.rs b/vhost/src/vhost_user/backend_req_handler.rs
index 07e01a9..14290bb 100644
--- a/vhost/src/vhost_user/backend_req_handler.rs
+++ b/vhost/src/vhost_user/backend_req_handler.rs
@@ -573,10 +573,9 @@ impl<S: VhostUserBackendReqHandler> BackendReqHandler<S> {
                 self.check_proto_feature(VhostUserProtocolFeatures::SHARED_OBJECT)?;
                 self.check_request_size(&hdr, size, hdr.get_size() as usize)?;
                 let msg = self.extract_request_body::<VhostUserSharedMsg>(&hdr, size, &buf)?;
-                let res = self.backend.get_shared_object(msg);
-                let reply_hdr = self.new_reply_header::<VhostUserEmpty>(&hdr, 0)?;
 
-                match res {
+                let reply_hdr = self.new_reply_header::<VhostUserEmpty>(&hdr, 0)?;
+                match self.backend.get_shared_object(msg) {
                     Ok(file) => {
                         self.main_sock.send_message(
                             &reply_hdr,
```